porsche-digital-lab-berlin-blockchain/catbox
# Feat/configs

## What does it do?
This pr makes all functions configurable so that the library can use any loom dappchain network and connect to ipfs using any valid ipfsConfig object.

It also implements importable preset configs for local development and external testing.

Lastly tests are all set to use local networks (local ipfs node and local plasmachain)

## How to test?
Firstly, make sure that [catbox-contracts](https://github.com/porsche-digital-lab-berlin-blockchain/catbox-contracts/pull/2) pr has been merged.

**on `catbox-contracts`:**
```sh
yarn loom:run
```
in another terminal:
```sh
yarn deploy
```
in yet another terminal (assuming you have ipfs installed, you need it if you don't):
```sh
ipfs daemon
```
**on this repo:**
```
yarn test
```
see that all tests run (and more quickly then previously, which used extdev network)
